src/core/ddsc/include/dds/ddsc/dds_public_qos.h

```diff
@@ -482,6 +482,23 @@ dds_qset_prop (
   const char * name,
   const char * value);
 
+/**
+ * @ingroup qos_setters
+ * @component qos_obj
+ * @brief Set value if the property shall be included in discovery or not.
+ *
+ * @param[in,out] qos - Pointer to a dds_qos_t structure that contains the property
+ * @param[in] name - Pointer to name of the property
+ * @param[in] propagate - Bool value to propagate or not
+ * 
+ * @returns - false if any of the arguments is invalid, the qos is not present in the qos object or there was no property found with the provided name
+ */
+DDS_EXPORT bool
+dds_qset_prop_propagate (
```

Comment:
I would instead add a function `dds_qset_prop_propagate(dds_qos_t *qos, const char *name, const char *value, bool propagate)` and define `dds_qset_prop` using that function: `dds_qset_prop(qos, name, value) { dds_qset_prop_propagate(qos, name, value, false); }`. I think that's a lot easier than having to set the properties by name and then set the propagation.